npc/events/christmas_2006.txt

```diff
@@ -0,0 +1,4165 @@
+			next;
+			mes "[Enjoy]";
+			mes "< ^0000ff5 Tree Knots^000000 >";
+			mes "< ^0000ff5 Wooden Blocks^000000 >";
+			mes "< ^0000ff1 Branch of Dead Tree^000000 >";
+			mes "Hurry!Bring it now!!";
+			disablenpc "Enjoy#06xmas2";
+			disablenpc "Happymerry#06xmas_02";
+			disablenpc "Christ#06xmas_02";
+			disablenpc "Mas#06mas_02";
+			disablenpc "Event#06xmas_02";
+			enablenpc "Switch#06christ_01";
+			close;
+		}
+	} else if (05christ_solo == 15) {
+		mes "[Enjoy]";
+		mes "It's useless complaining about the reality! let's make the better world for singles!!! Let's go for it!";
+		next;
+		mes "[Enjoy]";
+		mes "Till the day comes!!!";
+		mes "Till the day we become happy!";
+		mes "We won't lose our faith!! We won't give up!!";
+		mes "Let's enjoy ourselves till we feel better!";
+		05christ_solo = 0;
+		disablenpc "Enjoy#06xmas2";
+		enablenpc "Switch#06christ_01";
+		close;
+	} else {
+		mes "[Enjoy]";
+		mes "Who are you?!";
+		close;
+	}
+
+	OnGo:
+		disablenpc "Enjoy#06xmas2";
+		initnpctimer;
+		end;
+
+	OnStop:
+		enablenpc "Enjoy#06xmas2";
+		killmonster "prt_fild05", "Enjoy#06xmas2::OnMobDead";
+		stopnpctimer;
+		end;
+
+	OnInit:
+		disablenpc "Enjoy#06xmas2";
+		end;
+
+	OnTimer3000:
+		announce "Never cry for the past.", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer5000:
+		announce "Never wait for the unexpected phone call!!", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer7000:
+		announce "Never have a lingering affection for the love that never have started!!!", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer9000:
+		announce "Never break down being a lonely single!!!", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer11000:
+		announce "Never remember the sad memory!!!.", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer13000:
+		announce "...I am the happy single soldier!!!", bc_map, 0x66FFCC;
+		end;
+
+	OnTimer180000:
+		killmonster "prt_fild05", "Enjoy#06xmas2::OnMobDead";
+		announce "To all my single soldier friends~Merry christmas.", bc_map, 0x66FFCC;
+		donpcevent "Enjoy#06xmas2::OnStop";
+		end;
+
+	OnMobDead:
+		end;
+}
+
+prt_fild05,97,164,5	script	Happymerry#06xmas_02	4_M_LGTMAN,{
+	end;
+
+	OnInit:
+		disablenpc "Happymerry#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Happymerry#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,103,164,4	script	Christ#06xmas_02	4_M_PECOKNIGHT,{
+	end;
+
+	OnInit:
+		disablenpc "Christ#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Christ#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,103,159,1	script	Mas#06xmas_02	4_M_ALCHE_A,{
+	end;
+
+	OnInit:
+		disablenpc "Mas#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Mas#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,97,159,7	script	Event#06xmas_02	4_M_SITDOWN,{
+	end;
+
+	OnInit:
+		disablenpc "Event#06xmas_02";
+		end;
+
+	OnEnable:
+		enablenpc "Event#06xmas_02";
+		emotion ET_GO;
+		end;
+}
+
+prt_fild05,188,177,4	script	Happymerry#06xmas01	4_M_LGTMAN,{
+	if (05christ_solo == 9) {
+		mes "[Happymerry]";
```

Comment:
`.@name$ = strcharinfo(0);` to avoid the redundant command call

Reply:
Applied suggested fix.